### PR TITLE
docs: add a tutorial on how to write E2E tests

### PR DESCRIPTION
**Pull request by szymonlopaciuk**
## Description

Create an E2E test writing tutorial page in docs with a walk through how to write an E2E which harvests a non-core record from ArXiV and checks if it ends up in holdingpen as HALTED.

## Motivation and Context
Everyone can learn how to write E2E tests for INSPIRE.

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by szymonlopaciuk** [comment]
ci test please

**Comment by david-caro** [comment]
@Dinika test please

File: docs/e2e_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+.. code-block:: bash
+
+    $ docker inspect inspirenext_test-web-e2e.local_1 | grep '"IPAddress"'
+
+                "IPAddress": "",
+                    "IPAddress": "172.20.0.9",
+
+    $ sudo vim /etc/hosts
+
+And add a line add the bottom:
```
Comment (Dinika) [inline docs/e2e_tutorial.rst]: @szymonlopaciuk I know you're bit of a perfectionist, so to bug you I'm pointing out that there's a typo here.

Comment (szymonlopaciuk) [reply docs/e2e_tutorial.rst]: Thank you, @Dinika :smile:

**Comment by szymonlopaciuk** [comment]
ci merge please

**Pull request merged**